### PR TITLE
Added solution

### PR DESCRIPTION
**Pull request by AndrewPolishchuc**

### CODE REVIEW

File: src/main/java/core/basesyntax/ArrayList.java [hunk 1]
```diff
@@ -1,53 +1,108 @@
...
+        }
+    }
+
+    public T[] ensureCapacity() {
+        if (elementData.length - 1 - size >= ONE_ELEMENT) {
+            return elementData;
+        }
+        T[] newArrayData = (T[]) new Object[elementData.length + (elementData.length >> 1)];
+        System.arraycopy(elementData, 0, newArrayData, 0, size);
+        return newArrayData;
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/ArrayList.java:43]: can we make this method void and reassign  the result to element data inside?

File: src/main/java/core/basesyntax/ArrayList.java [hunk 2]
```diff
@@ -1,53 +1,108 @@
...
+        checkingIndex(index);
+        elementData[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkingIndex(index);
+        T removed = (T) elementData[index];
+        System.arraycopy(elementData, index + 1, elementData, index, size - index);
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/ArrayList.java:84]:
```suggestion
        System.arraycopy(elementData, index + 1, elementData, index, size - index -1);
```

File: src/main/java/core/basesyntax/ArrayList.java (hunk 1 above)
Comment (AndrewPolishchuc) [reply src/main/java/core/basesyntax/ArrayList.java:43]: Sure. Done)

File: src/main/java/core/basesyntax/ArrayList.java (hunk 2 above)
Comment (AndrewPolishchuc) [reply src/main/java/core/basesyntax/ArrayList.java:84]: Overlooked... Done)

File: src/main/java/core/basesyntax/ArrayList.java [hunk 3]
```diff
@@ -1,53 +1,108 @@
...
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0 ? true : false;
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/ArrayList.java:106]: try to simplify  that

File: src/main/java/core/basesyntax/ArrayList.java [hunk 4]
```diff
@@ -1,53 +1,108 @@
...
+        }
+    }
+
+    public void checkingIndex(int index) {
+        if (index >= size || index < 0) {
+            throw new ArrayIndexOutOfBoundsException("Out of bounds search index");
+        }
+    }
+
+    public void ensureCapacity() {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/ArrayList.java:37]: make them private and move to the end of class

File: src/main/java/core/basesyntax/ArrayList.java [hunk 5]
```diff
@@ -1,53 +1,108 @@
...
+        if (elementData.length - 1 - size >= ONE_ELEMENT) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/ArrayList.java:38]: try to change this condition without constant

File: src/main/java/core/basesyntax/ArrayList.java (hunk 4 above)
Comment (AndrewPolishchuc) [reply src/main/java/core/basesyntax/ArrayList.java]: Done.

File: src/main/java/core/basesyntax/ArrayList.java (hunk 3 above)
Comment (AndrewPolishchuc) [reply src/main/java/core/basesyntax/ArrayList.java]: Done.

File: src/main/java/core/basesyntax/ArrayList.java (hunk 5 above)
Comment (AndrewPolishchuc) [reply src/main/java/core/basesyntax/ArrayList.java]: Done.

**Pull request closed without merging**